File: dart/1007.dart
class Solution {
    int min(int a, int b){
        return (a < b) ? a : b;
    }

    int max(int a, int b){
        return (a > b) ? a : b;
    }

    int find(List<int> t, List<int> b, int v) {
        int ts = 0, bs = 0, n = t.length;
        for (int i = 0; i < n; i++) {
            if (t[i] == v)
                ts++;
            if (b[i] == v)
                bs++;
            else if (t[i] != v)
                return -1;
        }
        return min(min(ts, bs), n - max(ts, bs));
    }

    int minDominoRotations(List<int> t, List<int> b) {
        int r = t.length;
        for (int i = 1; i < 7; i++) {
            int v = find(t, b, i);
            if (v != -1)
                r = min(r, v);
        }
        if (r == t.length)
            return -1;
        return r;
    }
}

File: python3/1007.py
class Solution:
    def find(self, t, b, v):
        ts = 0
        bs = 0
        n = len(t)
        for i in range(n):
            if t[i] == v:
                ts += 1
            if b[i] == v:
                bs += 1
            elif t[i] != v:
                return -1
        return min(min(ts, bs), n - max(ts, bs))

    def minDominoRotations(self, t: List[int], b: List[int]) -> int:
        r = len(t)
        for i in range(1, 7):
            v = self.find(t, b, i)
            if v != -1:
                r = min(r, v)
        if r == len(t):
            return -1
        return r
